addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/handler/MySensorsThingHandler.java

```diff
@@ -0,0 +1,396 @@
+
+        registerListeners();
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        myGateway.removeEventListener(this);
+        super.dispose();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("MySensors Bridge Status updated to {} for device: {}", bridgeStatusInfo.getStatus().toString(),
+                getThing().getUID().toString());
+        if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)
+                || bridgeStatusInfo.getStatus().equals(ThingStatus.OFFLINE)) {
+
+            if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
+                registerListeners();
+            } else {
+                myGateway.removeEventListener(this);
+            }
+
+            // the node has the same status of the bridge
+            updateStatus(bridgeStatusInfo.getStatus());
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("Configuation update fo thing {}-{}: {}", nodeId, childId, configurationParameters);
+        super.handleConfigurationUpdate(configurationParameters);
+    }
+
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.trace("New state {} received for channel uid {}", newState, channelUID);
+        super.handleUpdate(channelUID, newState);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command {} received for channel uid {}", command, channelUID);
+        /*
+         * We don't handle refresh commands yet
+         *
+         */
+        if (command == RefreshType.REFRESH) {
+            return;
+        }
+
+        int intRequestAck = requestAck ? 1 : 0;
+
+        // just forward the message in case it is received via this channel. This is special!
+        if (channelUID.getId().equals(CHANNEL_MYSENSORS_MESSAGE)) {
+            if (command instanceof StringType) {
+                StringType stringTypeMessage = (StringType) command;
+                try {
+                    MySensorsMessage msg = MySensorsMessage.parse(stringTypeMessage.toString());
+                    myGateway.sendMessage(msg);
+                } catch (ParseException e) {
+                    logger.error("Invalid message to send", e);
+                }
+            }
+        } else {
+            MySensorsTypeConverter adapter = loadAdapterForChannelType(channelUID.getId());
+
+            if (adapter != null) {
+                logger.trace("Adapter {} found for type {}", adapter.getClass().getSimpleName(), channelUID.getId());
+
+                Integer type = adapter.typeFromChannelCommand(channelUID.getId(), command);
+
+                if (type != null) {
+                    logger.trace("Type for channel: {}, command: {} of thing {} is: {}", thing.getUID(), command,

```

Comment:
You're using the log levels in a proper way and contain nice amount of trace message if you really need to debug stuff 👍